### PR TITLE
add diagnostics for non-implemented data watchpoints

### PR DESCRIPTION
**Pull request by aap-sc**
Change-Id: If5031c6d8cea1bfcc34bb65fd766f232498ed7ea

### CODE REVIEW

File: src/target/riscv/riscv.c [hunk 1]
```diff
@@ -1293,6 +1293,14 @@ static void trigger_from_watchpoint(struct trigger *trigger,
 
 int riscv_add_watchpoint(struct target *target, struct watchpoint *watchpoint)
 {
+	// NOTE: typeof is needed because of upstream OpenOCD bug. This should be
+	// replaced by WATCHPOINT_IGNORE_DATA_VALUE_MASK once it is available
+	// See: https://review.openocd.org/c/openocd/+/7840
+	if (watchpoint->mask != ~(typeof(watchpoint->mask))0) {
+		LOG_ERROR("watchpoints on data values are not implemented");
```
Comment (timsifive) [inline src/target/riscv/riscv.c]:
```suggestion
		LOG_TARGET_ERROR(target, "watchpoints on data values are not implemented");
```

**Review by timsifive — CHANGES_REQUESTED** [review]
Looks good overall.

File: src/target/riscv/riscv.c (hunk 1 above)
Comment (aap-sc) [reply src/target/riscv/riscv.c]: addressed

File: src/target/riscv/riscv.c [hunk 2]
```diff
@@ -1293,6 +1293,14 @@ static void trigger_from_watchpoint(struct trigger *trigger,
 
 int riscv_add_watchpoint(struct target *target, struct watchpoint *watchpoint)
 {
+	// NOTE: typeof is needed because of upstream OpenOCD bug. This should be
+	// replaced by WATCHPOINT_IGNORE_DATA_VALUE_MASK once it is available
+	// See: https://review.openocd.org/c/openocd/+/7840
+	if (watchpoint->mask != ~(typeof(watchpoint->mask))0) {
+		LOG_TARGET_ERROR(target, "watchpoints on data values are not implemented");
```
Comment (MarekVCodasip) [inline src/target/riscv/riscv.c]: One tiny thing, I would capitalize the first letter in the "watchpoints" error log

**Review by MarekVCodasip — DISMISSED** [review]
But that is a minor thing, giving approve.

File: src/target/riscv/riscv.c (hunk 2 above)
Comment (aap-sc) [reply src/target/riscv/riscv.c]: addressed

**Review by MarekVCodasip — APPROVED** [review]

**Review by timsifive — APPROVED** [review]

**Pull request merged**